### PR TITLE
Update GEP-1742 to use GEP-2257 Duration type

### PR DESCRIPTION
**Pull request by frankbu**
**What type of PR is this?**

/kind gep

**What this PR does / why we need it**:
Completing the update that @kflynn started in #2316.
This PR includes Flynn's changes.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by kflynn — APPROVED** [review]
Looks great, thanks!

**Comment by robscott** [comment]
Thanks @frankbu!

/check-cla
/ok-to-test

**Review by robscott — COMMENTED** [review]
Thanks @frankbu, one small thing but otherwise LGTM.

File: geps/gep-1742.md [hunk 1]
```diff
@@ -388,8 +390,12 @@ type HTTPRouteRule struct {
...
 
 // HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
-// Timeout values are formatted like 1h/1m/1s/1ms as parsed by Golang time.ParseDuration
-// and MUST BE >= 1ms or 0 to disable (no timeout).
+// Timeout values are formatted like 1h/1m/1s/1ms, as specified in GEP-2257,
+// and MUST BE >= 1ms or "0s" to disable (no timeout).
+//
+// +kubebuilder:validation:XValidation:message="request timeout value must be greater than or equal to 1ms",rule="!(has(self.request) && duration(self.request) != duration('0s') && duration(self.request) < duration('1ms'))"
+// +kubebuilder:validation:XValidation:message="backendRequest timeout value must be greater than or equal to 1ms",rule="!(has(self.backendRequest) && duration(self.backendRequest) != duration('0s') && duration(self.backendRequest) < duration('1ms'))"
+// +kubebuilder:validation:XValidation:message="backendRequest timeout cannot be longer than request timeout",rule="!(has(self.request) && has(self.backendRequest) && duration(self.request) != duration('0s') && duration(self.backendRequest) > duration(self.request))"
```
[the remark this answers is not in the document]
Comment (frankbu) [reply geps/gep-1742.md:398]: Maybe I'm missing something, but I think it allows 0s with this part of the rule: `&& duration(self.request) != duration('0s')`. Anyway, I can remove these rules from here if you'd prefer we just add them in #2013 instead.

[the remark this answers is not in the document]
Comment (frankbu) [reply geps/gep-1742.md:398]: If we merge this as is, I will take the TODO to make sure they are updated if we end up needing to change them at all in the API.

[the remark this answers is not in the document]
Comment (robscott) [reply geps/gep-1742.md:398]: You might be right, I'd feel better if we introduced the CEL validation along with some proper go tests so we can confirm we're getting the right error message(s) with the right inputs. For example, this particular error message from the validation above feels wrong: "request timeout value must be greater than or equal to 1ms".

[the remark this answers is not in the document]
Comment (frankbu) [reply geps/gep-1742.md]: > I'd feel better if we introduced the CEL validation along with some proper go tests so we can confirm we're getting the right error message(s) with the right inputs.

OK, removed for now.

> For example, this particular error message from the validation above feels wrong: "request timeout value must be greater than or equal to 1ms".

I wondered about that, but my thinking is that 0s is an infinite duration, so just saying the timeout needs to be >= 1ms covers the 0 case too 😃

**Comment by robscott** [comment]
Thanks @frankbu!

/lgtm
/approve

**Pull request merged**